listenbrainz/webserver/static/js/src/personal-recommendations/PersonalRecommendationsModal.tsx

```diff
@@ -0,0 +1,263 @@
+        artist_name: getArtistName(recordingToPersonallyRecommend),
+        track_name: getTrackName(recordingToPersonallyRecommend),
+        release_name: recordingToPersonallyRecommend!.track_metadata
+          .release_name,
+        recording_mbid: getRecordingMBID(recordingToPersonallyRecommend!),
+        recording_msid: getRecordingMSID(recordingToPersonallyRecommend!),
+        users,
+        blurb_content: blurbContent,
+      };
+      try {
+        const status = await APIService.submitPersonalRecommendation(
+          currentUser.auth_token,
+          currentUser.name,
+          metadata
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            "You personally recommended a track!",
+            `${metadata.artist_name} - ${metadata.track_name}`
+          );
+          this.setState({ blurbContent: "" });
+        }
+      } catch (error) {
+        this.handleError(error, "Error while personally recommending");
+      }
+    }
+  };
+
+  render() {
+    const { recordingToPersonallyRecommend } = this.props;
+    if (!recordingToPersonallyRecommend) {
+      return null;
+    }
+    const { blurbContent, users, suggestions } = this.state;
+    const { track_name } = recordingToPersonallyRecommend.track_metadata;
+    const { artist_name } = recordingToPersonallyRecommend.track_metadata;
+    const { APIService, currentUser } = this.context;
+    return (
+      <div
+        className="modal fade"
+        id="PersonalRecommendationModal"
+        tabIndex={-1}
+        role="dialog"
+        aria-labelledby="PersonalRecommendationModalLabel"
+        data-backdrop="static"
+      >
+        <div className="modal-dialog" role="document">
+          <form className="modal-content">
+            <div className="modal-header">
+              <button
+                type="button"
+                className="close"
+                data-dismiss="modal"
+                aria-label="Close"
+                onClick={() => {
+                  this.setState({
+                    users: [],
+                    blurbContent: "",
+                    suggestions: [],
+                  });
+                  // update the followers list
+                  APIService.getFollowersOfUser(currentUser.name)
+                    .then((response) => {
+                      this.setState({ followers: response.followers });
+                    })
+                    .catch((error) => {
+                      this.handleError(error, "Error while fetching followers");
+                    });
+                }}
+              >
+                <span aria-hidden="true">&times;</span>
+              </button>
+              <h4 className="modal-title" id="PersonalRecommendationModalLabel">
+                Personally recommend this recording
+              </h4>
+            </div>
+            <div className="modal-body">
+              {users!.map((user) => {
```

Comment:
Why the need for the `!`?
That's usually a "bad smell" (as programmers call it) and point to the type not being well defined.

I see in `PersonalRecommendationModalState` the arrays are also described as being potentially `null`, but looking at the code they will never be so and should simply be an array.
For example in the constructor you explicitly set them as an empty array, and in all your setState calls you are careful and always them them to arrays too.
That way the need for the `!` will go away.